### ISSUE TITLE
All Page Scan Does Not Function

### ISSUE THREAD

**Issue by nickspurry** [OpaitSoftware/TesseractStudio.Net#1]
When performing an OCR scan with page range selected to "All", Tesseract Studio does not perform the OCR scan and returns an error dialog saying "Error: Failed to recognize page". Error is repeatable on multiple documents, possibly every document (occurs on every document I have tried).

Scanning with page range selected to "Current" works as expected. 

Tesseract Studio Version: 1.1.5.36939
OS: Windows 10 Professional Version 1809
OCR Language: Eng
Spell Checker Dictionary: None/en_US (same behavior with both settings)
Parallel Processes: Automatic
Overwrite Existing OCR: Selected/Deselected (same behavior with both settings)

**Comment by FarhadKhalafi** [comment]
Are you processing multipage TIFF or PDF files? Any possibility of sharing a failing document?

**Comment by nickspurry** [comment]
Multipage PDF documents. I can't share documents publicly due to copyright constraints.

**Comment by nickspurry** [comment]
This error no longer occurs on my machine after updating to the latest version 1.2.0.41757.

**Comment by OpaitSoftware** [comment]
One cause of this error was the size of the PDF file. For larger files, the system was running out of memory and aborting. We have implemented a streaming interface to eliminate all un-necessary image caching in the next version to be released shortly. The result is that the overall memory consumption has been reduced drastically and made only dependent on the number of processing cores. We have tested the new design with PDF files of 2,500 images with limited increase in peak memory consumption as compared with 8 pages on a processor with 8 cores.
